tests/distributed/test_rpc.py

```diff
@@ -188,15 +188,16 @@ def test_multi_client():
     os.environ['DGL_DIST_MODE'] = 'distributed'
     generate_ip_config("rpc_ip_config_mul_client.txt", 1, 1)
     ctx = mp.get_context('spawn')
-    pserver = ctx.Process(target=start_server, args=(10, "rpc_ip_config_mul_client.txt"))
+    num_clients = 20
+    pserver = ctx.Process(target=start_server, args=(num_clients, "rpc_ip_config_mul_client.txt"))
     pclient_list = []
-    for i in range(10):
+    for i in range(num_clients):
```

Comment:
Will num_client = 20 hang the test if you do not fix the bug?

Reply:
yes, sometimes hang.

Reply:
actually, this testcase hangs even `num_clients=10` if `cython-enabled` dgl is installed. In our CI test env, `cython` is always not enabled, so we don't hit this before. I need to figure out a way to enable `cython` in CI, so that we could catch such error. Namely, we'd better test on released DGL.